### PR TITLE
Explode operation for KeyTable

### PR DESCRIPTION
**Pull request by jigold**

### CODE REVIEW

File: src/main/scala/org/broadinstitute/hail/keytable/KeyTable.scala [hunk 1]
```diff
@@ -420,4 +420,39 @@ case class KeyTable(rdd: RDD[(Annotation, Annotation)], keySignature: TStruct, v
 
     KeyTable(newRDD, keySignature, valueSignature)
   }
+
+  def explode(columnNames: Array[String]): KeyTable = {
```
Comment (cseed) [inline src/main/scala/org/broadinstitute/hail/keytable/KeyTable.scala]: I think I would have written explode to take a single column, and then have the version that takes multiple columns to call the single-column version in a loop.  You deal with the looping in a number of places below and you can simplify the code overall by pulling the out to a single loop outside.

**Comment by jigold** [comment]
@cseed: back to you

File: python/pyhail/keytable.py [hunk 2]
```diff
@@ -361,6 +361,7 @@ def select(self, field_names):
         except Py4JJavaError as e:
             self._raise_py4j_exception(e)
 
+<<<<<<< de99211b52c45fd1d587bcfca5eebc9c19b77665
```
Comment (cseed) [inline python/pyhail/keytable.py]: You left in some diff/rebase chaff.

**Comment by jigold** [comment]
@cseed back to you

File: src/main/scala/org/broadinstitute/hail/keytable/KeyTable.scala [hunk 3]
```diff
@@ -489,4 +489,37 @@ case class KeyTable(rdd: RDD[(Annotation, Annotation)], keySignature: TStruct, v
       },
       schema.asInstanceOf[StructType])
   }
+
+  def explode(columnName: String): KeyTable = {
```
Comment (cseed) [inline src/main/scala/org/broadinstitute/hail/keytable/KeyTable.scala]: This is much cleaner!  I like.

File: python/pyhail/keytable.py [hunk 4]
```diff
@@ -387,3 +387,51 @@ def to_dataframe(self, expand=True, flatten=True):
...
+    def explode(self, column_names):
+        """Explode columns of this KeyTable.
+
+        The explode operation unpacks the elements in a column of type ``Array`` or ``Set`` into its own row.
+        If an empty ``Array`` or ``Set`` is exploded, the entire row is removed from the ``KeyTable``.
+
+        **Examples**
+
+        Assume ``kt`` is a :py:class:`.KeyTable` with two columns: c1 and c2.
+        The types of each column are ``String`` and ``Array[Int]``.
```
Comment (cseed) [inline python/pyhail/keytable.py]: Sorry, I'm not being clear.  I thought your old example was good, I just thought you needed a simpler one, too.  I'd include both!

**Pull request merged**